src/Ng/CommandHelpers.cs

```diff
@@ -45,86 +114,113 @@ private static void TraceRequiredArgument(string name)
             Trace.TraceError("Required argument \"{0}\" not provided", name);
         }
 
-        public static string GetSource(IDictionary<string, string> arguments)
+        private static void TraceMissingArgument(string name)
         {
-            string value;
-            if (!arguments.TryGetValue("-source", out value))
+            Console.WriteLine("Argument \"{0}\" not provided", name);
+            Trace.TraceWarning("Argument \"{0}\" not provided", name);
+        }
+
+        private static void TryGetArgument(IDictionary<string, string> arguments, string searchArg, out string value, bool required = false)
```

Comment:
>void [](start = 23, length = 4)

in general, when a method is called TryGetXYZ we expect it to have a certain structure. Meaning, it returns a value using out parameter, and returns a bool, saying if it was successful or not. Although we do something a bit different here, with the throw, I still think it's a good idea to return a bool in case we dont throw.